airgun/views/contenthost.py

```diff
@@ -0,0 +1,233 @@
+import re
+
+from widgetastic.widget import (
+    Checkbox,
+    ParametrizedView,
+    Select,
+    Text,
+    TextInput,
+    View,
+)
+from widgetastic_patternfly import BreadCrumb, Button
+
+from airgun.views.common import (
+    AddRemoveResourcesView,
+    AddRemoveSubscriptionsView,
+    BaseLoggedInView,
+    LCESelectorGroup,
+    SatTab,
+    SatTabWithDropdown,
+    SatTable,
+    SearchableViewMixin,
+    TaskDetailsView,
+)
+from airgun.widgets import (
+    ActionsDropdown,
+    ConfirmationDialog,
+    EditableEntry,
+    EditableEntryCheckbox,
+    EditableEntrySelect,
+    ReadOnlyEntry,
+    Search,
+)
+
+
+class ContentHostsView(BaseLoggedInView, SearchableViewMixin):
+    title = Text("//h2[contains(., 'Content Hosts')]")
+    export = Text(
+        ".//a[contains(@class, 'btn')][contains(@href, 'content_hosts.csv')]")
+    register = Text(".//button[@ui-sref='content-hosts.register']")
+    actions = ActionsDropdown(".//div[contains(@class, 'btn-group')]")
+    dialog = ConfirmationDialog()
+    table = SatTable(
+        './/table',
+        column_widgets={
+            0: Checkbox(locator="./input[@type='checkbox']"),
+            'Name': Text('./a'),
+        }
+    )
+
+    @property
+    def is_displayed(self):
+        return self.browser.wait_for_element(
+            self.title, exception=False) is not None
+
+
+class ContentHostDetailsView(BaseLoggedInView):
+    breadcrumb = BreadCrumb()
+    unregister = Text(".//button[@ng-click='openModal()']")
+    dialog = ConfirmationDialog()
+
+    @property
+    def is_displayed(self):
+        breadcrumb_loaded = self.browser.wait_for_element(
+            self.breadcrumb, exception=False)
+        return (
+            breadcrumb_loaded
+            and self.breadcrumb.locations[0] == 'Content Hosts'
+            and len(self.breadcrumb.locations) > 1
+        )
+
+    @View.nested
+    class details(SatTab):
+        # Basic information
+        name = EditableEntry(name='Name')
+        uuid = ReadOnlyEntry(name='UUID')
+        description = EditableEntry(name='Description')
+        type = ReadOnlyEntry(name='Type')
+        katello_agent = ReadOnlyEntry(name='Katello Agent')
+        virtual_guests = ReadOnlyEntry(name='Virtual Guests')
+        registered_through = ReadOnlyEntry(name='Registered Through')
+        # Subscriptions
+        subscription_status = ReadOnlyEntry(name='Subscription Status')
+        details = ReadOnlyEntry(name='Details')
+        auto_attach = EditableEntryCheckbox(name='Auto-Attach')
```

Comment:
maybe add link to `Run Auto-Attach`

Reply:
Can't see such link there:
<img width="417" alt="image" src="https://user-images.githubusercontent.com/11719030/44355266-d2a69080-a4b4-11e8-8d12-aa7525417f2b.png">

any details of where can i find it?